--- include/optional	2026-05-31 16:03:09.442969310 +0400
+++ include-patched/optional	2026-05-31 18:56:27.551103596 +0400
@@ -2285,9 +2285,11 @@
     ranges::enable_borrowed_range<optional<_Tp&>> = true;
 #endif
 
+#if __glibcxx_format_ranges
   template<typename _Tp>
     inline constexpr range_format
       format_kind<optional<_Tp>> = range_format::disabled;
+#endif
 #endif // __cpp_lib_optional_range_support
 
 #undef _GLIBCXX_USE_CONSTRAINTS_FOR_OPTIONAL
