diff '--color=auto' -urN include/stddef.h include-patched/stddef.h
--- include/stddef.h	2024-09-05 07:27:19.393557253 +0200
+++ include-patched/stddef.h	2024-09-05 07:55:12.109152021 +0200
@@ -430,7 +430,7 @@
      use __float128 here; that is only available on some
      architectures, but only on i386 is extra alignment needed for
      __float128.  */
-#ifdef __i386__
+#if defined(__i386__) && !defined(__clang__)
   __float128 __max_align_f128 __attribute__((__aligned__(__alignof(__float128))));
 #endif
 } max_align_t;
